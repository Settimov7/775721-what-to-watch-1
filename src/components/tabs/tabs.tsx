import * as React from 'react';

type Tab = {
  title: string,
  component: React.ReactElement,
}

interface Props {
  tabs: Tab[];
  activeItem: number;
  onChange: (index: number) => void;
}

export const Tabs = (props: Props) => {
  const {tabs, activeItem, onChange} = props;

  const activeComponent = tabs[activeItem].component;

  const renderNavigation = () => (
    <nav className="movie-nav movie-card__nav">
      <ul className="movie-nav__list">
        {tabs.map((tab, index) => {
          const {title} = tab;
          const isActive = index === activeItem;

          return (
            <li key={title} className={`movie-nav__item ${isActive && 'movie-nav__item--active'}`}>
              <a className="movie-nav__link"
                onClick={(evt) => {
                  evt.preventDefault();

                  onChange(index)
                }}>
                {title}
              </a>
            </li>
          );
        })}
      </ul>
    </nav>
  );

  return(
    <React.Fragment>
      <nav className="movie-nav movie-card__nav">
        <ul className="movie-nav__list">
          {renderNavigation()}
        </ul>
      </nav>

      {activeComponent}
    </React.Fragment>
  );
};
