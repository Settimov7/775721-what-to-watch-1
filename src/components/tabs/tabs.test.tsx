import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {Tabs} from './tabs';

const MockComponent = () => <div />;

it(`Tabs correctly renders`, () => {
  const props = {
    tabs: [
      {
        title: `Tab 1`,
        component: <MockComponent />,
      },
      {
        title: `Tab 2`,
        component: <MockComponent />,
      },
      {
        title: `Tab 3`,
        component: <MockComponent />,
      },
    ],
    activeItem: 0,
    onChange: jest.fn(),
  };

  const tabs = renderer.create(
      <Tabs {...props}/>
  ).toJSON();

  expect(tabs).toMatchSnapshot();
});
