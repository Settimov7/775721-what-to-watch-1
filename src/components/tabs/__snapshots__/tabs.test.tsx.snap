// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs correctly renders 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <nav
        className="movie-nav movie-card__nav"
      >
        <ul
          className="movie-nav__list"
        >
          <li
            className="movie-nav__item movie-nav__item--active"
          >
            <a
              className="movie-nav__link"
              onClick={[Function]}
            >
              Tab 1
            </a>
          </li>
          <li
            className="movie-nav__item false"
          >
            <a
              className="movie-nav__link"
              onClick={[Function]}
            >
              Tab 2
            </a>
          </li>
          <li
            className="movie-nav__item false"
          >
            <a
              className="movie-nav__link"
              onClick={[Function]}
            >
              Tab 3
            </a>
          </li>
        </ul>
      </nav>
    </ul>
  </nav>,
  <div />,
]
`;

exports[`Tabs correctly renders 1`] = `
Array [
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <nav
        className="movie-nav movie-card__nav"
      >
        <ul
          className="movie-nav__list"
        >
          <li
            className="movie-nav__item movie-nav__item--active"
          >
            <a
              className="movie-nav__link"
              onClick={[Function]}
            >
              Tab 1
            </a>
          </li>
          <li
            className="movie-nav__item false"
          >
            <a
              className="movie-nav__link"
              onClick={[Function]}
            >
              Tab 2
            </a>
          </li>
          <li
            className="movie-nav__item false"
          >
            <a
              className="movie-nav__link"
              onClick={[Function]}
            >
              Tab 3
            </a>
          </li>
        </ul>
      </nav>
    </ul>
  </nav>,
  <div />,
]
`;
